cql3/functions/vector_similarity_fcts.cc

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2025-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+ */
+
+#include "vector_similarity_fcts.hh"
+#include "types/types.hh"
+#include "types/vector.hh"
+#include "exceptions/exceptions.hh"
+
+namespace cql3 {
+namespace functions {
+namespace {
+
+// You should only use this function if you need to preserve the original vectors and cannot normalize
+// them in advance.
```

Comment:
I still think a comment should appear somewhere in this file explaining that the bizarre (for me...) choices of scaling and even squaring of the similarities is meant to be compatible with the exact formulas from Cassandra, and that we have tests to confirm this is actually the case.